spec/index.html

```diff
@@ -3162,6 +3159,54 @@ <h3>Double Circle</h3>
   </section>
 </section>
 
+<section id="canonical-quads" class="appendix">
+  <h2>A Canonical form of N-Quads</h2>
+
+  <p>This section defines a canonical form of N-Quads which has
+    a completely specified layout.
+    The grammar for the language is the same.</p>
+
+  <p>Canonical N-Quads updates and extends
+    <a data-cite="N-TRIPLES#canonical-ntriples">Canonical N-Triples</a> in [[N-TRIPLES]]
+    to include <code><a data-cite="N-QUADS#grammar-production-graphLabel">graphLabel</a></code>.</p>
+
+  <p>While the N-Quads syntax allows choices for the representation and layout of RDF data,
+    the canonical form of N-Quads provides a unique syntactic representation of any quad.
+    Each code point
+    can be represented by only one of
+    <code><a data-cite="N-QUADS#grammar-production-UCHAR">UCHAR</a></code>,
+    <code><a data-cite="N-QUADS#grammar-production-ECHAR">ECHAR</a></code>,
+    or unencoded character,
+    where the relevant production allows for a choice in representation.
+    Each quad is represented entirely on a single line with specified white space.</p>
+
+  <p>Canonical N-Quads has the following additional constraints on layout:</p>
+  <ul>
+    <li>White space MUST NOT be used except after
+      <code>subject</code>,
+      <code>predicate</code>, 
+      <code>object</code>,
+      and <code>graphLabel</code>,
+      any of which MUST be a single space (<code>U+0020</code>).</li>
```

Comment:
```suggestion
      each of which MUST be a single space (<code>U+0020</code>).</li>
```

Reply:
I'll note that we went through several rounds of those wording already in the N-Quads document.